### DISCUSSION TITLE
Importing aligned TS from EMPIAR

### DISCUSSION THREAD

**Discussion by asarnow** [nextpyp/.github#94] [Q&A]
There are a number of EMPIAR depositions containing only aligned tiltseries and .tlt (and .xf, .local, .erase) files.

To use these data in PYP do we need to generate MDOC files to import the tilts angles? If so I can write a script to do so and post it here.

**Comment by tofushan** [comment]
Sorry, we currently do not support importing these metadata into nextPYP. You will need to start the pre-processing from scratch. Additionally, you will need to re-align the tilt-series even though they are already aligned. But you can enter `90.0` degree for `Tilt-axis angle (degrees)` when importing the raw data. This is what we use for EMPIAR-10304 where the tilt-series are also aligned.

**Reply by asarnow** [reply]
I got all that, but what about the actual tilt angles? Do I need to generate dummy MDOC files or write them into the MRC headers or something?

**Reply by tofushan** [reply]
If the tilt-series are already aligned, they should be single mrc file per tilt-series. You can follow the parameters for the Classification tutorial. You will also need to rename `.tlt` to `.rawtlt`, and put them together with the tilt-series mrc files.

**Comment by asarnow** [comment]
I figured it out:

* Tilt angles are read from a .rawtlt file, matching the .mrc TS stack (filter raw data directory for *.mrc files)
* ~.rawtlt files cannot have any leading whitespace~
  * ~IMO this is a bug as the aligned TS can be provided with the final .tlt (refined angles) instead of a .rawtlt and these frequently have leading whitespace~
* .order files are also needed

What is the format for the .order file? The ones from the 10304 tutorial are a sequential list of [0,40] interval, but the reference says the TS are dose-symmetric. Should the file have the dose-symmetric order instead, if the aligned TS is ordered by tilt?

**Reply by tofushan** [reply]
Great! 

Yes, the order file is sorted by tilt angles. If your data is dose-symmetric, it should look something like `..., 4, 1, 0, 2, 3, ...` where each line has one number instead of comma.

**Reply by asarnow** [reply]
Thanks! I was wrong about the leading whitespace, though, it was really that I needed to run a new import block after adding the .rawtlt files.

**Reply by asarnow** [reply]
However, there is another bug, that the number of tilts is not necessarily being counted correctly.

<img src="https://github.com/nextpyp/.github/assets/4052606/c693dd15-6be3-4944-a90c-aac07203b074" width="50%" />

<img src="https://github.com/nextpyp/.github/assets/4052606/aec98f8e-f31d-401b-a3f7-796cff2b7323" width="50%" />

**Reply by asarnow** [reply]
OK, not a bug. The error message is indicates .rawtlt but it could be either .rawtlt or .order with the mismatched number of tilts. Correcting the .order files worked.

(So technically - there is a bug, but it's just the error message).

**Reply by tofushan** [reply]
Glad to see it works! We can fix the error message to take both .rawtlt and .order into account.

**Reply by abartesaghi** [reply]
This was addressed in version [0.6.4](https://nextpyp.app/files/pyp/0.6.4/docs/index.html).
